core/src/main/java/org/elasticsearch/transport/RequestHandlerRegistry.java

```diff
@@ -52,8 +55,14 @@ public Request newRequest() {
             return requestFactory.get();
     }
 
-    public TransportRequestHandler<Request> getHandler() {
-        return handler;
+    public void processMessageReceived(Request request, TransportChannel channel) throws Exception {
+        final Task task = taskManager.register(channel.channelType(), action, request);
```

Comment:
I would prefer for all requests to create a task because I think it will provide us with really valuable insight about what's going on in the cluster, but I will add an option to opt out.